src/components/ToDoComponent/TodoComponent.tsx

```diff
@@ -3,42 +3,54 @@ import React, { useEffect, useState } from 'react';
 import Card from './Card';
 import data from './Data';
 import { TodoStyles } from './Styles/TodoStyles';
+import Task from './taskType';
 
 const TodoComponent = () => {
   const [tasks, setTasks] = useState(data);
-  const [position, setpostion] = useState(1);
+  const [disabled, setDisabled] = useState(false);
+  const [changed, setChanged] = useState(false);
 
-  const changeCardFunction = (id: any) => {
-    const newData = tasks.filter((item) => item.id == id);
-    tasks.shift();
-    tasks.push(newData[0]);
-    position !== tasks[tasks.length - 1].id
-      ? setpostion(tasks[position + 1].id)
-      : setpostion(tasks[0].id);
+  useEffect(() => {
+    setTasks(tasks);
+  }, [changed, tasks]);
+
+  const changeCardFunction = () => {
```

Comment:
My believe there is no need of `useEffect here:
```js
  const changeCardFunction = (id: number) => {
    if (tasks.length !== 1) {
      const task = tasks[0];
      const redorderdTasks = tasks.filter((t) => t.id !== id);
      setTasks([...redorderdTasks, task]);
    }
  };
  ```

Reply:
+1

Reply:
hey @shmbajaj I've tried this solution, this creates some problems, so if a user swipes that component multiple times it leads to duplication of that component

Reply:
so that is why I was using shift to remove that component and push it back to the end of array

Reply:
Though is `tasks` should be read-only and `setTasks` is enought for rerendering.
```js
    const reorderedTasks = [...tasks];
    const t = reorderedTasks.shift() as Task;
    setTasks([...reorderedTasks, t]);
```

Reply:
> hey @shmbajaj I've tried this solution, this creates some problems, so if a user swipes that component multiple times it leads to duplication of that component and we get an error `Encountered two children with the same key`

Can you guide to reproduce this error.